cmd/docker/docker.go

```diff
@@ -419,10 +419,11 @@ func areSubcommandsSupported(cmd *cobra.Command, details versionDetails) error {
 		if cmdVersion, ok := curr.Annotations["version"]; ok && versions.LessThan(details.CurrentVersion(), cmdVersion) {
 			return fmt.Errorf("%s requires API version %s, but the Docker daemon API version is %s", cmd.CommandPath(), cmdVersion, details.CurrentVersion())
 		}
-		if ost, ok := curr.Annotations["ostype"]; ok && ost != details.ServerInfo().OSType {
-			return fmt.Errorf("%s is only supported on a Docker daemon running on %s, but the Docker daemon is running on %s", cmd.CommandPath(), ost, details.ServerInfo().OSType)
+		si := details.ServerInfo()
```

Comment:
Whoops, and I just realised I introduced a regression here; this was deliberately put inline to allow it to be executed conditionally (see https://github.com/docker/cli/commit/b39739123b845f872549e91be184cc583f5b387c / https://github.com/docker/cli/pull/2424)

Opening a PR to fix that 😅